lib/vsc/filesystem/oceanstor.py

```diff
@@ -512,11 +524,83 @@ def get_namespace_info(self, namespace):
             self.log.raiseException(errmsg, OceanStorOperationError)
             return None
 
+    def _is_bucket(self, namespace):
+        """
+        Check if namespace is a bucket in OceanStor
+
+        @returns: boolean corresponding to the bucket attribute. None if undetermined due to permissions.
+        """
+        oceanstor_namespaces = self.list_namespaces()
+        known_namespaces = [namespace for acc in oceanstor_namespaces for namespace in oceanstor_namespaces[acc]]
```

Comment:
shouldn't we also cache this?

Reply:
Depends how often we need this, recalculate this? From what I understand, this will be recalculate a lot?

Reply:
Done in https://github.com/vub-hpc/vsc-filesystem-oceanstor/pull/26/commits/258659270f8128c207ab18b64d761f6e650ab52a
* `ocenastor_namespaces` is now a 1 level dict with keys per namespace name
* `oceanstor_account_namespaces` is now the 2 level dict with namespaces per account